### ISSUE TITLE
How to use do_lower_case=False when using a cased custom Bert model?

### ISSUE THREAD

**Issue by BerndStromberg** [explosion/spacy-transformers#98]
Hey,

I don't want the pipeline to lowercase everything in `TransformersWordPiecer` when I'm loading my custom trained Bert model.

There is a `do_lower_case` param in Huggingface's Transformers for Bert Tokenization (see: https://github.com/huggingface/transformers/blob/700331b5ece63381ad1b775fc8661cf3ae4493fd/transformers/tokenization_bert.py). **How can I set this param via spacy transformers?** 

I tried `TransformersWordPiecer.from_pretrained(nlp.vocab, path, do_lower_case=False)` but it failed. Doing the same for `TransformersLanguage(...)` failed, too.

We trained a german bert model and word casing is important (for our applications). Right now everything gets lowered when looking at `doc._.trf_word_pieces_`.

**Comment by BerndStromberg** [comment]
Today I had some time to investigate. Here is the solution for those who have also stumbled across this issue:

**tldr:** see code at the end.

The spacy-transformers tutorial tells us to load the WordPiecer by calling 
`TransformersWordPiecer.from_pretrained(nlp.vocab, "path/to/bert-model-dir")`. If we are having a look on what it is actually doing, we see the following code:
```
    # code located at spacy_transformers/pipeline/wordpiecer.py
    @classmethod
    def from_pretrained(cls, vocab, trf_name, **cfg):
        model = get_tokenizer(trf_name).from_pretrained(trf_name)
        return cls(vocab, model=model, trf_name=trf_name, **cfg)
```

Param `model `is the tokenizer itself and therefore responsible for how your tokenization works. All the other stuff is basically some spacy-transformers handling for wrapping Huggingface's Transformers, serialization, etc. 

Okay, so we need to pass `do_lower_case` here: `get_tokenizer(trf_name).from_pretrained(trf_name, do_lower_case=False)`. But it will never reach it, because the only thing that gets passed is `trf_name`. In other words: using `**cfg` has no effect on `get_tokenizer().from_pretrained()`. 

Solution: Don't use `TransformersWordPiecer.from_pretrained()` and let it be responsible for loading the model. Instead initialize TransformersWordPiecer via object constructor and pass the model by yourself. Code:

    trf_path = "path/to/your-bert-model"
    transformer = TransformersLanguage(trf_path, meta={"lang": "xx"})
    transformer.add_pipe(transformer.create_pipe("sentencizer"))

    # these two lines are changed; model is initialized by us and passed as param:
    model = get_tokenizer("bert").from_pretrained(trf_path, do_lower_case=False)
    twp = TransformersWordPiecer(transformer.vocab, model=model, trf_name=trf_path)
    transformer.add_pipe(twp)

    transformer.add_pipe(TransformersTok2Vec.from_pretrained(transformer.vocab, trf_path))

FYI: If you get `"Unsupported transformers config name:"` error,  then it's because your `trf_path` is missing the word `"bert"`. The `get_tokenizer()` method decides which tokenizer you need based on that. That's why I'm passing bert directly.

**Comment by svlandeg** [comment]
Sorry for the late follow-up, but happy to see you got it working! I'll assume this issue can be closed - thanks for sharing the solution in such detail!
